packages/components/src/navigator/router/route.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Internal dependencies
+ */
+import matchPath from './match-path';
+import RouterContext from './router-context';
+
+/**
+ * WordPress dependencies
+ */
+import { Children, Component, createElement } from '@wordpress/element';
+
+function isEmptyChildren( children ) {
+	return Children.count( children ) === 0;
+}
+
+/**
+ * The public API for matching a single path and rendering.
+ */
+class Route extends Component {
+	render() {
+		/* eslint-disable no-nested-ternary */
```

Comment:
Should we get rid of the nested ternaries? I find them very difficult to read.

Reply:
Definitely! There are many more substantial issues that I wanted to take care of, but I'd definitely like for this to be fixed before merging.